core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricTokenParser.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import spray.json._
+
+class InvalidJwtTokenException(message: String) extends Exception(message)
+class JwtTokenExpiryMissingException(message: String) extends Exception(message)
+class FabricTokenParser(JWToken: String) {
```

Comment:
Im not sure you need to make these classes, seems like you really just want a small utility function here. Youl can also just use RuntimeExceptions or InvalidArgumentExceptions instead of making your own

Reply:
THis comment still applies

Reply:
The original plan was the same as you have recommended here. But, then since FabricTokenParser is used to check the validity of the token. Most importantly, the expiration. Thus, I decided to group the invalid token exception into one that is expired and rest into one bucket. This will help me debug in future. So, let's keep this as is for now.